src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Security/RoleNames.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+
+namespace OrchardCore.Infrastructure.Security
+{
+    /// <summary>
+    /// Defines well-known default and system role names.
+    /// </summary>
+    public static class RoleNames
+    {
+        private static readonly HashSet<string> s_systemRoleNames = new(StringComparer.OrdinalIgnoreCase)
+        {
+            Anonymous,
+            Authenticated,
+        };
+
+        public const string Administrator = "Administrator";
+        public const string Anonymous = "Anonymous";
+        public const string Authenticated = "Authenticated";
+        public const string Author = "Author";
+        public const string Contributor = "Contributor";
+        public const string Editor = "Editor";
+        public const string Moderator = "Moderator";
+
+        public static IEnumerable<string> SystemRoles => s_systemRoleNames;
```

Comment:
I think these helper should stay where they were and keep role name in here.

Reply:
I have changed that because the `RoleHelper` class had very few references internally and I don't expect many uses outside of the internal projects. Also I think it makes sense to bundle system roles and the role names together and I thought `RoleNames` would be easier to discover, additionally because I changed the namespace to match the one of `IRole` too.

Reply:
`RoleNames` should be constant variable whereas `RoleHelper` is a class for helper function. If someone is using the RoleNames, they'll have a breaking change

Reply:
Sorry, I don't understand what you mean. Do you want to move `RoleNames` inside `RoleHelper`? Breaking is the removal of `RoleHelper`, but I don't think that matters. It contains one static field and has 4 internal references. I don't expect any external reference.

I messed up the namespace of `RoleNames` anyway. It was meant to be in `OrchardCore.Security` side-by-side with `IRole` and `Role`.

Reply:
@gvkries 
I suggest to keep the existing RoleHelper in addition to the new RoleNames. 

I also, think you should to only replace role name with the new constants in this PR. Then using a follow up PR maybe suggest the casing changes. This way the logic of changing the casing will be much easier to follow.

Reply:
No need to take this, it was just a suggestion to make things easier in future. I'm going to close this PR and provide the fix only.